### PR TITLE
Dev Environment Setup

### PR DESCRIPTION
**Pull request by RealJasomo**
## Description of Work
Generated the initial boilerplate for the project and got various configuration files setup
## Acceptance Criteria

- Initial boilerplate for NextJS, Tailwind, Prisma, NextAuth, and tRPC
- eslint and prettier configured
- Github actions pipeline created
- Docker-compose for local test services (Postgres database)
- Onboarded Deepsource SCA

## How to Test

- Follow setups in app/README.md and verify that the project can be run from your computer

### List of Changes:

- t3-stack boilerplate
- Readme created for setting up the project
- docker files created

### CODE REVIEW

**Comment by RealJasomo** [comment]
Should be working, but fails due to needing the secrets set under Settings > Security/Secrets >Actions

File: app/prisma/schema.prisma
Comment (DavidPurdy1) [inline app/prisma/schema.prisma:1]: This is cool, makes things a lot easier.

**Review by DavidPurdy1 — CHANGES_REQUESTED** [review]
Looks pretty good. I would like the instructions to be a little more in depth because there was a couple things that came up. I would really like 1-2 more people to review this since it is the setup we will be using for this project and this is a lot of files (and I'm not too experienced what some of these boilerplate files do). @rhit-daviscl @cryerca

File: app/.gitignore
Comment (RealJasomo) [reply app/.gitignore]: That was part of autogenerated code

File: app/.prettierrc.json
Comment (RealJasomo) [reply app/.prettierrc.json]: That's a good point, changed it to reflect that.

File: app/next-env.d.ts
Comment (RealJasomo) [reply app/next-env.d.ts:1]: This is a triple-slash directive. You can find out more about them [here](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html). Basically this file is for module type definitions and is part of the boilerplate

File: app/README.md
Comment (RealJasomo) [reply app/README.md:28]: This secure key is just a random secret for authentication. It can be anything you want since it is local

File: app/README.md
Comment (RealJasomo) [reply app/README.md]: Good catch, I copied the wrong repo lol

File: .deepsource.toml
Comment (RealJasomo) [reply .deepsource.toml:19]: Not sure, this was autogenerated by deep source so I left it as is.

File: app/README.md
Comment (rhit-daviscl) [inline app/README.md:33]: These will be for when we are confirming our AWS connections, correct?
Asked in Teams, but asked again for GitHub.

**Review by rhit-daviscl — COMMENTED** [review]
Overall looks good to me. David covered the areas that could be shrunken down and most the others are added automatically, to a degree. All good from me, but lets have others try and set their dev up before we approve and merge.

File: app/next-env.d.ts
Comment (DavidPurdy1) [reply app/next-env.d.ts:1]: Okay thanks

File: .deepsource.toml
Comment (DavidPurdy1) [reply .deepsource.toml:19]: Okay, I don't care a ton either, just trying to get things to look nice.

File: app/README.md
Comment (DavidPurdy1) [reply app/README.md:17]: Okay I was able to get the dev environment setup on my windows machine working fine. Considering you've been able to on a Linux VM and my Linux Machine is currently broken, I would say it is probably an issue with my machine. The only other thing that would be nice is if we could get picture steps for the readme, but not super important.

**Review by DavidPurdy1 — APPROVED** [review]
Alright, I am happy with the dev environment setup. I am going to go ahead an approve and it's up to you if you want to check with everyone else regarding any issues. Nice work.

**Pull request merged**
